fsm/builder.go

```diff
@@ -59,11 +68,12 @@ func (b *Builder) Build() (*FSM, error) {
 	}
 
 	m := &FSM{
-		currentState: b.start,
+		currentState: stater.GetState(),
```

Comment:
Hrm.  I'd actually contemplate taking this in a different direction.  Instead of adding a new argument to `Build()`, what if we created a new method in `fsm` called `JSONEncoder` that an implementation must satisfy.  For instance:

```go
package fsm
func JSONEncoder(fsm *FSM, w io.Writer) (int64, error) {
  // ...
}

func JSONDecoder(r io.Reader) (*FSM, error) {
  // ...
}
```

Or something like that.  ?